### PR TITLE
Update test cases for attestations API.

### LINKED ISSUE #3389: Ensure attestation APIs have good test cases
Many test cases have been added sporadically for oe_evidence API set. This issue is to ensure all oe_evidence APIs are covered. Cases include:
1. oe_get_evidence() can generate all supported combinations of:
-  format IDs
-  flags
-  custom_claims
-  custom_claims_size (< or > 64 bytes)
2. oe_verify_evidence() can verify good and bad evidence and returns expected claims (especially report_data vs. custom_claims)

### PR DESCRIPTION
**Pull request by qiucwang**
Fixes #3389 

Checked all public attestation APIs and macros are covered by test cases.

Added test case for large custom claims buffer. A 65-byte custom claims buffer are tested. It worked with the evidence format `OE_FORMAT_UUID_SGX_ECDSA_P256` as it will be hashed into a 64-byte buffer and placed in the SGX report data field. And it failed with the evidence formats `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT & Quote` as the claim buffer will be directly placed in the SGX report data field and the claim buffer size exceeded the 64 bytes limit.

Added test case for `oe_verify_evidence()`, verifying the `oe_report` generated by `oe_get_report()` and the `SGX_quote` extracted from `oe_report` to make sure `oe_verify_evidence()` can verify the evidence not generated by `oe_get_evidence()`

Put custom claims buffer in `oe_get_evidence(OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT, ...)` and `oe_get_evidence(OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE, ...)`, which are retrieved and verified in the `claims[OE_CLAIM_SGX_REPORT_DATA]` by `oe_verify_evidence()`.

Signed-off-by: Qiucheng Wang <qiucwang@microsoft.com>

### CODE REVIEW

File: tests/attestation_plugin/plugin/tests.c [hunk 1]
```diff
@@ -29,6 +29,11 @@
 #include "tests.h"
 
 uint8_t test_claims[TEST_CLAIMS_SIZE] = "This is a sample test claims buffer";
+// Should succeeded for oe_evidence oe_but failed for oe_report and raw sgx
```
Comment (shnwc) [inline tests/attestation_plugin/plugin/tests.c:32]:
```suggestion
// Should succeed for oe_evidence oe_but fail for oe_report and raw sgx
```

**Comment by qiucwang** [comment]
bors try

File: tests/attestation_plugin/plugin/tests.c (hunk 1 above)
Comment (qiucwang) [reply tests/attestation_plugin/plugin/tests.c:32]: Thanks for pointing out!

**Comment by qiucwang** [comment]
bors retry

**Pull request merged**